tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -672,3 +673,29 @@ def val_dataloader(self):
         enable_model_summary=False,
     )
     trainer.fit(model)
+
+
+@pytest.mark.parametrize("num_dataloaders", [1, 2, 11])
+def test_log_metrics_only_include_metrics_from_concerned_dataloader(num_dataloaders, tmpdir):
+    class TestModel(BoringModel):
+        def test_step(self, batch, batch_idx, dataloader_idx=0):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            self.log("fake_test_acc", loss)
+            return {"y": loss}
+
+        def test_epoch_end(self, *_) -> None:
+            pass
+
+    test = RandomDataset(32, 2)
+    test_dataloaders = [DataLoader(test, batch_size=1)] * num_dataloaders
+
+    model = TestModel()
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
```

Comment:
max_epochs has no purpose for test

```suggestion
    trainer = Trainer(default_root_dir=tmpdir)
```